trieste/acquisition/rule.py

```diff
@@ -1434,11 +1448,11 @@ def state_func(
                     }
                 else:
                     _datasets = None
+                acquisition_space = TaggedMultiSearchSpace(subspaces, self._tags)
                 points = self._rule.acquire(acquisition_space, models, _datasets)
 
-            # We may modify the regions in filter_datasets later, so return a copy.
-            state_ = BatchTrustRegion.State(copy.deepcopy(acquisition_space))
-            return state_, tf.reshape(points, [-1, len(self._subspaces), points.shape[-1]])
+            state_ = BatchTrustRegionState(subspaces, self._tags)
+            return state_, tf.reshape(points, [-1, len(self._init_subspaces), points.shape[-1]])
```

Comment:
Nitpick: probably slightly clearer to use `subspaces` as this is what the surrounding code is referring to.
 
```suggestion
            return state_, tf.reshape(points, [-1, len(subspaces), points.shape[-1]])
```